velox/type/DecimalUtil.h

```diff
@@ -146,40 +147,40 @@ class DecimalUtil {
   }
 
   template <typename TInput, typename TOutput>
-  inline static std::optional<TOutput> rescaleWithRoundUp(
+  inline static Status rescaleWithRoundUp(
```

Comment:
This is PR is a bit large and complicated. Would it be possible to extract this particular change into a separate PR to make review easier?

Reply:
Extract this change to https://github.com/facebookincubator/velox/pull/8465.